rasa/nlu/selectors/response_selector.py

```diff
@@ -338,21 +349,31 @@ def _full_response(self, label: Dict[Text, Any]) -> Optional[Dict[Text, Any]]:
             the text but also buttons, images, ...
         """
         for key, responses in self.responses.items():
-            for response in responses:
-                if hash(response.get(TEXT, "")) == label.get("id"):
-                    return response
-        return None
+            if self.train_on_text:
```

Comment:
this function isn't only used training, this property should get a better name

Reply:
Changed it to `use_text_as_label`.